lib/spark.js

```diff
@@ -240,6 +244,83 @@ export default class Spark {
   }
 }
 
+function getRetrievalUrl (protocol, address, cid) {
+  if (protocol === 'http') {
+    const baseUrl = multiaddrToHttpUri(address)
+    return `${baseUrl}/ipfs/${cid}?dag-scope=block`
+  }
+
+  const searchParams = new URLSearchParams({
+    // See https://github.com/filecoin-project/lassie/blob/main/docs/HTTP_SPEC.md#dag-scope-request-query-parameter
+    // Only the root block at the end of the path is returned after blocks required to verify the specified path segments.
+    'dag-scope': 'block',
+    protocols: protocol,
+    providers: address
+  })
+  return `ipfs://${cid}?${searchParams.toString()}`
+}
+
+/**
+ * @param {string} cid
+ * @param {Uint8Array} carBytes
+ */
+async function verifyContent (cid, carBytes) {
+  let reader
+  try {
+    reader = await CarBlockIterator.fromBytes(carBytes)
+  } catch (err) {
+    throw Object.assign(err, { code: 'CANNOT_PARSE_CAR_BYTES' })
+  }
+
+  for await (const block of reader) {
+    if (block.cid.toString() !== cid.toString()) {
+      throw Object.assign(
+        new Error(`Unexpected block CID ${block.cid}. Expected: ${cid}`),
+        { code: 'UNEXPECTED_CAR_BLOCK' }
+      )
+    }
+
+    await validateBlock(block)
+  }
+}
+
+function mapErrorToStatusCode (err) {
+  // 90x codes for multiaddr parsing errors
```

Comment:
I believe codes until 599 are part of the standard, so we could start with 600 here

Reply:
not important though

Reply:
I think this is actually a good idea. We can use 7xx for multiaddr parsing errors, 8xx for retrieval errors, and 9xx for content verification errors.